Sources/Web3Core/EthereumNetwork/Request/APIRequest+Methods.swift

```diff
@@ -8,22 +8,116 @@
+        }
+
+        for value in object.values {
+            if let result = spelunkData(value: value) {
+                return result
+            }
+            return nil
+        }
+    }
+    if let array = value as? [Any] {
+        for e in array {
+            if let result = spelunkData(value: e) {
+                return result
+            }
+            return nil
+        }
+    }
+
+    // Might be a JSON string we can further descend...
+    if let string = value as? String, let data = string.data(using: .utf8) {
+        let json = try? JSONSerialization.jsonObject(with: data)
+        return spelunkData(value: json)
+    }
+
+    return nil
+}
+
 extension APIRequest {
     public static func sendRequest<Result>(with provider: Web3Provider, for call: APIRequest) async throws -> APIResponse<Result> {
-        let request = setupRequest(for: call, with: provider)
-        return try await APIRequest.send(uRLRequest: request, with: provider.session)
+        try await send(call.call, parameter: call.parameters, with: provider)
     }
 
-    static func setupRequest(for call: APIRequest, with provider: Web3Provider) -> URLRequest {
+    static func setupRequest(for body: RequestBody, with provider: Web3Provider) -> URLRequest {
         var urlRequest = URLRequest(url: provider.url, cachePolicy: .reloadIgnoringCacheData)
         urlRequest.setValue("application/json", forHTTPHeaderField: "Content-Type")
         urlRequest.setValue("application/json", forHTTPHeaderField: "Accept")
-        urlRequest.httpMethod = call.method.rawValue
-        urlRequest.httpBody = call.encodedBody
+        urlRequest.httpMethod = "POST"
+        urlRequest.httpBody = body.encodedBody
         return urlRequest
     }
 
-    public static func send<Result>(uRLRequest: URLRequest, with session: URLSession) async throws -> APIResponse<Result> {
+    public static func send<Result>(_ method: String, parameter: [Encodable], with provider: Web3Provider) async throws -> APIResponse<Result> {
+        let body = RequestBody(method: method, params: parameter)
+        let uRLRequest = setupRequest(for: body, with: provider)
+
+        let data: Data
+        do {
+            data = try await send(uRLRequest: uRLRequest, with: provider.session)
+        } catch Web3Error.rpcError(let error) {
+            let responseAsString = try checkError(method: method, error: error)
+            guard let LiteralType = Result.self as? LiteralInitiableFromString.Type,
+                  let literalValue = LiteralType.init(from: responseAsString),
+                  let result = literalValue as? Result else {
+                throw Web3Error.dataError
+            }
+            return APIResponse(id: 2, result: result)
+        }
+
+        /// This bit of code is purposed to work with literal types that come in ``Response`` in hexString type.
+        /// Currently it's just `Data` and any kind of Integers `(U)Int`, `Big(U)Int`.
```

Comment:
```suggestion
        /// Checks if `Result` type can be initialized from HEX-encoded bytes.
        /// If it can - we attempt initializing a value of `Result` type. 
```